File: 프로그래머스/0/181931. 등차수열의 특정한 항만 더하기/등차수열의 특정한 항만 더하기.java
class Solution {
    public int solution(int a, int d, boolean[] included) {
        int answer = 0;
        
        for(int i = 0; i < included.length; i++) {
            if(included[i]) {
                answer += a + d * i;
            }
        }
        
        return answer;
    }
}

// 다른 풀이
return IntStream.range(0, included.length)
                .map(idx -> included[idx] ? a+(idx*d) : 0)
                .sum();

File: 프로그래머스/0/181931. 등차수열의 특정한 항만 더하기/등차수열의 특정한 항만 더하기.py
def solution(a, d, included):
    answer = 0
    
    for i, include in enumerate(included):
        if include :
            answer += a + d * i
    
    return answer
    
# 다른 풀이
answer = 0
for i in range(len(included)):
    answer += (a + d * i) * int(included[i])
return answer

# 다른 풀이
return sum(a + i * d for i, f in enumerate(included) if f)
